### PR TITLE
Cms/kancer ezeroglu/netlify cms/docs fields/custom widgets

### PR DESCRIPTION
**Pull request by KancerEzeroglu**
**Summary**

**Test plan**

**A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: website/content/docs/custom-widgets.md [hunk 1]
```diff
@@ -198,3 +195,276 @@ You can also return a promise from `isValid`. While the promise is pending, the
...
+
+if (typeof window !== 'undefined') {
+  window.Control = Control
+  window.Preview = Preview
+}
+
+export { Control, Preview }
+```
+
+6. Change the content of `package.json` as below:
```
Comment (erezrokah) [inline website/content/docs/custom-widgets.md:290]: Instead of copy pasting the entire file, I think we should split it into steps that explain the process.
For example, we should have a "setup wepback" after:
https://github.com/netlify/netlify-cms/pull/4146/files#diff-71cf0183902b4785a350655ed78e3fb2R215

that says, "In order to build React components we need to set up a build step. We'll be using Webpack. Please run the following command to install the required dependencies..."

Then another step to setting up the actual Webpack config, then we can explain the creation of the index file to export the control component and preview component.

To summarize, describe the process of generating the different files and the reason behind it.

File: website/content/docs/custom-widgets.md [hunk 2]
```diff
@@ -198,3 +195,276 @@ You can also return a promise from `isValid`. While the promise is pending, the
...
+npm run build
+```
+
+8. Finally, if you're sure things are tested and working, publish!
+
+```javascript
+npm publish
+```
+
+### [](https://github.com/netlify/netlify-cms-widget-starter#deploying-a-live-demo)Deploying a live demo
```
Comment (erezrokah) [inline website/content/docs/custom-widgets.md:460]: We can drop this part as we're no longer in the context of a git repo.

**Review by erezrokah — CHANGES_REQUESTED** [review]
Thanks @KancerEzeroglu, great progress.
A few more comments and I think we're missing setting up the locale example site:
https://github.com/netlify/netlify-cms-widget-starter/tree/master/dev

File: website/content/docs/custom-widgets.md (hunk 1 above)
Comment (KancerEzeroglu) [reply website/content/docs/custom-widgets.md]: I have made changes. 
But I couldn't find how can I install "peerDependencies" from the command. ("peerDependencies": {},)
Should we add it manually to package.json?
I have searched tons of documents but [they say](https://stackoverflow.com/questions/51293822/webpack-cli3-0-8-requires-a-peer-of-webpack4-x-x-but-none-is-installed) if you install the webpack, peerDependencies will fill automatically.
But when I try to install all dependencies from scratch, it did not happen for me.
It should be like that:
"peerDependencies": {
    "react": "^16"
  },
Could you help me?

Comment (erezrokah) [reply website/content/docs/custom-widgets.md]: Thanks @KancerEzeroglu. I appreciate the effort. We can add it manually.

File: website/content/docs/custom-widgets.md (hunk 2 above)
Comment (KancerEzeroglu) [reply website/content/docs/custom-widgets.md]: I have dropped it, fyi

File: website/content/docs/custom-widgets.md (hunk 1 above)
Comment (KancerEzeroglu) [reply website/content/docs/custom-widgets.md]: I've made changes, it is ready for review. fyi

**Review by erezrokah — APPROVED** [review]
Thanks @KancerEzeroglu.
Great work following up.
I just made a small heading change so the new section will show up in the sidebar.
